core/src/main/java/hudson/cli/DeleteBuildsCommand.java

```diff
@@ -31,14 +31,14 @@
 import java.util.HashSet;
 import java.util.List;
 import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * Deletes builds records in a bulk.
  *
  * @author Kohsuke Kawaguchi
  */
-@Restricted(DoNotUse.class) // command implementation only
+@Restricted(NoExternalUse.class) // command implementation only
```

Comment:
What seems to be missing is a restriction which means “allowed to be used only inside this type”. There is no exact analogue in the Java language, since `javac` even relaxes `private` for use from other types in the same compilation unit (cf. [JEP-181](http://openjdk.java.net/jeps/181)), but an equally reasonable restriction would be “allowed to be used only in the same compilation unit”.

Reply:
Agreed, it looks like with the new changes `DoNotUse` cannot be used if the class has any inner or anonymous classes inside of it. I think using  `NoExternalUse` for now is totally fine.